nodejs/aws-infra/experimental/clusterAutoScaling.ts

```diff
@@ -0,0 +1,566 @@
+        const cluster = args.cluster;
+
+        let launchConfiguration: ClusterAutoScalingLaunchConfiguration;
+
+        // Use the autoscaling config provided, otherwise just create a default one for this cluster.
+        if (args.launchConfiguration) {
+            launchConfiguration = args.launchConfiguration;
+        }
+        else {
+            launchConfiguration = new ClusterAutoScalingLaunchConfiguration(
+                name, args.launchConfigurationArgs || { cluster }, parentOpts);
+        }
+
+        this.instance = new aws.cloudformation.Stack(name, {
+            ...args,
+            name: launchConfiguration.stackName,
+            templateBody: getCloudFormationTemplate(
+                name,
+                launchConfiguration.instance.id,
+                cluster.network.subnetIds,
+                args.templateParameters || {}),
+        }, parentOpts);
+
+        this.cluster = cluster;
+        this.launchConfiguration = launchConfiguration;
+    }
+}
+
+(<any>ClusterAutoScalingGroup).doNotCapture = true;
+
+// TODO[pulumi/pulumi-aws/issues#43]: We'd prefer not to use CloudFormation, but it's the best way to implement
+// rolling updates in an autoscaling group.
+function getCloudFormationTemplate(
+    instanceName: string,
+    instanceLaunchConfigurationId: pulumi.Output<string>,
+    subnetIds: pulumi.Input<string>[],
+    parameters: pulumi.Input<TemplateParameters>): pulumi.Output<string> {
+
+    const subnetIdsArray = pulumi.all(subnetIds);
+    return pulumi.all([subnetIdsArray, instanceLaunchConfigurationId, parameters])
+                 .apply(([subnetIdsArray, instanceLaunchConfigurationId, parameters]) => {
+
+    const minSize = parameters.minSize !== undefined ? parameters.minSize : 2;
+    const maxSize = parameters.maxSize !== undefined ? parameters.maxSize : 100;
+
+    return `
+    AWSTemplateFormatVersion: '2010-09-09'
+    Outputs:
+        Instances:
+            Value: !Ref Instances
+    Resources:
+        Instances:
+            Type: AWS::AutoScaling::AutoScalingGroup
+            Properties:
+                Cooldown: 300
+                DesiredCapacity: ${minSize}
+                HealthCheckGracePeriod: 120
+                HealthCheckType: EC2
+                LaunchConfigurationName: "${instanceLaunchConfigurationId}"
+                MaxSize: ${maxSize}
+                MetricsCollection:
+                -   Granularity: 1Minute
+                MinSize: ${minSize}
+                VPCZoneIdentifier: ${JSON.stringify(subnetIdsArray)}
+                Tags:
+                -   Key: Name
+                    Value: ${instanceName}
+                    PropagateAtLaunch: true
+            CreationPolicy:
+                ResourceSignal:
+                    Count: ${minSize}
+                    Timeout: PT15M
+            UpdatePolicy:
+                AutoScalingRollingUpdate:
+                    MaxBatchSize: 1
+                    MinInstancesInService: ${minSize}
```

Comment:
note: i'm providing the same level of customization that cloud currently provides.  However, we should likely make every single thing in this template something you can parameterize.  i.e. MaxBatchSize, Timeout, PropagateAtLaunch, etc.